### ISSUE TITLE
Sign In, Sign Out, Profile Page

### ISSUE THREAD

**Issue by buckyroberts** [thenewboston-developers/Website#315]
- [x] Give users the ability to sign in with GitHub. The **Sign In** button should send users directly to the GitHub Sign In page. 
- [x] Include a **Create Account** (with GitHub) in the marketing page as well, this page will take you directly to the GitHub Create your account page. Once the user creates an account, we will redirect to thenewboston website signed in. 
- [x] When the user signs into thenewboston, redirect them to their profile page.
- [x] Once you're signed into thenewboston website, we do not need to display the Sign In & Create Account button on the top nav, we only need to display your avatar on the right of the Download button.
- [x] Selecting on the avatar will display a dropdown of 2 buttons, Your Profile & Sign Out.
- [x] For mobile, if you are signed in, inside the top nav, we are going to include options for 'Your Profile' & 'Sign Out' 
- [x] The profile page will display the users avatar (pull in their GitHub profile image), the users Display Name (filled out by the user), the users role & team, Slack Name (filled out by the user), GitHub username (link to their GitHub), account number (with QR code), & all the tasks they completed. 
- [x] Display the Team Lead tag on the profile if they are one.
- [x] When a user is viewing their own profile, there will be a 'Edit Profile' icon. Here they can edit their Display Name, Slack Name, & Account Number. 
- [x] Just to note, the avatar has a white stroke (2px) around it, so it doesn't blend in with the background color.
- [x] On the 1366px screens, the user info block & the task completed block thats been designed is the max width of those containers, for smaller screens, the blocks will respond according to the tablet/mobile
- [x] On the leaderboard, the users github names will now link to the users profile.
- [x] On the tasks page, clicking on the users avatar will link to the users profile. 
- [x] For mobile specifically, on the Tasks Completed section, we can remove the Repository section & the Completed section. We will move Total Earnings up to the top and we will include Total Tasks on the left. 
- [x] For mobile, when the user edits their profile, the edit profile should take up the whole screen. The standard pattern is, Save button will be on the top right, close button will be on the top left. 
- [x] We are going to update the mobile top nav by putting the Get Started & Community options into the dropdown. (Currently all the options are displayed without selecting the dropdown) We are going to do this because now that there are more options on the top nav, we need to collapse the options so the newer options are more prominent. 
- [x] For mobile, it is a little difficult to click the nav items with your thumb since they are so close together, so we are updating them to have 24px padding in-between. 

https://www.figma.com/file/ChyT88EmP2Rmzx35I9erCL/Marketing-Site?node-id=1353%3A1915

**Comment by angle943** [comment]
this one is not ready for engineering yet

**Comment by angle943** [comment]
prereq: https://github.com/thenewboston-developers/Website/issues/461

**Comment by jamessspanggg** [comment]
> * [ ]  On the leaderboard, the users github names will now link to the users profile.
> * [ ]  On the tasks page, clicking on the users avatar will link to the users profile.

It seems like these two points are not done, should I create a separate PR and take on this issue? @thesanjeevsharma @Hristijan95

**Comment by jamessspanggg** [comment]
Just to check, are all these tasks done?

**Comment by Rajputusman** [comment]
@jamessspanggg #1123 is still opened so should I wait until it is closed before moving it to Paid?

**Comment by buckyroberts** [comment]
Closing since this is out of date. Majority is done but waiting on some backend changes. Any further changes needed we can split up into a separate ticket.
